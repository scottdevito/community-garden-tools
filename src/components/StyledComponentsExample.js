import React, { Component } from 'react';
import styled from 'styled-components';
import { CTAPrimaryRed, Card } from '../reusable-styles/reusable-styles';

type Props = {
  colorProp?: string,
};

class StyledComponentsExample extends Component<Props> {
  render() {
    return (
      <StyledComponentsWrapper>
        <Box colorProp="#9575CD">
          <StyledComponentsExampleHeader>
            Styled Components
          </StyledComponentsExampleHeader>
          <ul>
            <li>Install styled components</li>
          </ul>
          <CTAPrimaryRed>This is a reusable button</CTAPrimaryRed>
        </Box>
      </StyledComponentsWrapper>
    );
  }
}

export default StyledComponentsExample;

const StyledComponentsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const Box = styled(Card)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: ${props => props.colorProp};
  color: #fff;
`;

const StyledComponentsExampleHeader = styled.h1``;
