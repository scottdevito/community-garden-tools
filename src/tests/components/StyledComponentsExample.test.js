import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

import StyledComponentsExample from '../../components/StyledComponentsExample';

// Snapshot test
it('renders correctly', () => {
  const tree = renderer.create(<StyledComponentsExample />).toJSON();
  expect(tree).toMatchSnapshot();
});
