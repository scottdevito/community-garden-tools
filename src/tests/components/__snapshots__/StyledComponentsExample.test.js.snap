// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-ifAKCX idxgcQ"
>
  <div
    className="sc-EHOje VdJQc"
  >
    <h1
      className="sc-bZQynM bOrPFB"
    >
      Styled Components
    </h1>
    <ul>
      <li>
        Install styled components
      </li>
    </ul>
    <button
      className="sc-htpNat kRUzAD"
    >
      This is a reusable button
    </button>
  </div>
</div>
`;
